src/mono/netcore/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.Mono.cs

```diff
@@ -224,7 +218,8 @@ public sealed class AssemblyBuilder : Assembly
         [MethodImplAttribute(MethodImplOptions.InternalCall)]
         private static extern void UpdateNativeCustomAttributes(AssemblyBuilder ab);
 
-        internal AssemblyBuilder(AssemblyName n, string? directory, AssemblyBuilderAccess access, bool corlib_internal)
+        [DynamicDependency(nameof(pktoken))] // Automatically keeps all previous fields too due to StructLayout
```

Comment:
Why do we need DynamicDependency at all? The class is marked `StructLayout.Sequential` and this annotation is on a constructor - linker is going to keep all the fields anyway if it sees the constructor is needed.

Reply:
Technically we don't right now but I wanted to make this dependency more explicit. We could also fix linker to keep only used fields for struct layout classes.

Reply:
I'm not sure I understand how an explicit DynamicDependency on what appears to be an arbitrary field achieves that - I think we'll find that part quite confusing when we look at this code again 5 years from now.

Reply:
If we want to be explicit, can we instead use the attribute that says to keep all fields?